ext-apiserver/gcp-deployer/CONTRIBUTING.md

```diff
@@ -0,0 +1,153 @@
+2. Make sure your user can execute docker commmands (without sudo). This is a way to test it:
+```bash
+$ docker run hello-world
+
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+...
+```
+
+### Install OpenSSL
+
+Install [OpenSSL](https://www.openssl.org/source/) on your machine. Please note that this is just temporary. We are working to remove this dependency. See [Issue](https://github.com/kubernetes/kube-deploy/issues/591)
+
+### Install APIServer-Builder (Optional)
+
+If you need to rebuild container image for the extension APIServer and Controller Manager, you will need to install [APIServer-builder](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/installing.md)
+
+## Fetch Source Code
+
+1. Fork [kube-deploy repo](https://github.com/kubernetes/kube-deploy). If it's your first time forking, please take a look at [GitHub Repo instructions](https://help.github.com/articles/fork-a-repo/). The general [Kubernetes GitHub workflow](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md) is helpful here too if you're getting started.
+
+2. Clone Repo Locally
+```bash
+$ cd $GOPATH/src/k8s.io/
+$ git clone https://github.com/<GITHUB_USERNAME>/kube-deploy.git
+```
+
+## Build
+
+```bash
+$ cd $GOPATH/src/k8s.io/kube-deploy/ext-apiserver/gcp-deployer/
+$ go build
+```
+
+This will create a binary `gcp-deployer` in the same directory. You can use that binary to manage a GCP cluster.
+
+## Developing
+
+When making changes to the machine controller, it's generally a good idea to delete any existing cluster created with an older version of the cluster-api.
+
+```bash
+$ ./gcp-deployer delete
+```
+
+After making changes to the controllers or the actuator, you need to follow these two steps:
+
+1. Rebuild the machine-controller image. Also change `machineControllerImage` in `cloud/google/pods.go` to the new image path (make sure the version in the Makefile and `pods.go` match if you want to use the new image). Then, rebuild and push the image.
+
+	```bash
+	$ cd $GOPATH/src/k8s.io/kube-deploy/ext-apiserver
+	$ apiserver-boot build container --image gcr.io/$(GCLOUD_PROJECT)/apiserver-controller:$(VERSION) --generate=false
+	```
+
+NOTE: that the image will be pushed to `gcr.io/$(GCLOUD_PROJECT)/apiserver-controller`. Image storage is a billable resource.
+
+2. Rebuild gcp-deployer
+
+	```bash
+    $ cd $GOPATH/src/k8s.io/kube-deploy/ext-apiserver/gcp-deployer/
+	$ go build
+	```
+
+The new `gcp-deployer` will have your changes.
+
+## Testing
+
+We do not have unit tests or integration tests currently. For any changes, it is recommended that you test a create-edit-delete sequence using the new machine controller image and the new `gcp-deployer` binary.
+
+1. Generate machines configuration file.
+
+This step is necessary to include the project name (as configured in Google Cloud SDK) in the yaml file.
+
+```bash
+$ ./generate-yaml.sh
```

Comment:
the generate-yaml.sh script needs to be copied over into this directory, along with machines.yaml.template.